framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -529,8 +529,7 @@ protected void addInformationFromPreconditions(
                                 false);
                 info.insertValue(expr, annotation);
             } catch (FlowExpressionParseException e) {
-                // report errors here
-                analysis.checker.report(e.getResult(), methodTree);
```

Comment:
I'm not sure what you mean with 'In this case...`.
If we move these checks to the BaseTypeVisitor, all type systems would benefit from them, not just those that use flow refinement. That would seem like a good thing. Do you agree with this or not?